### PR TITLE
Support preview releases

### LINKED ISSUE #697: Support creating beta releases
We should be able to create beta/pre-release releases. These should be published to GitHub but marked as pre-release, but not published to Chocolatey.

### PR DESCRIPTION
**Pull request by ErikMogensen**
Fixes #697 

Since Assembly version and file version don't allow letters this PR adds setting the Informational file version, which allows letters. Then we can have 5.3.4-beta1.

### CODE REVIEW

File: GetNumberVersion.ps1 [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Set-StrictMode -Version 3
+
+if ($Version -match "^\d+\.\d+\.\d+$") {
+	return $Version
+}
+
+if (-not ($Version -match "^\d+\.\d+\.\d+-.{1,12}$")) {
+	throw "The version must start with three numbers separated by dots. This may be " + `
+		"followed by a '-' and a word. That word may not be longer than 12 characters. For instance, 4.3.22-beta."
```
Comment (SeanFeldman) [inline GetNumberVersion.ps1:16]: > That word may not be longer than 12 characters

Not according to [SemVer](https://semver.org/).

Comment (ErikMogensen) [reply GetNumberVersion.ps1]: In the FAQ part it says: 

> Does SemVer have a size limit on the version string?
> 
> No, but use good judgment. A 255 character version string is probably overkill, for example. Also, specific systems may impose their own limits on the size of the string.

Anyway, I changed it to use the impressive regex mentioned on https://semver.org.

Comment (SeanFeldman) [reply GetNumberVersion.ps1]: That's what my comment was about. The 12 characters constrain seemed unnecessary 🙂

**Comment by SeanFeldman** [comment]
It's not my intent to stall this PR. Just doesn't feel right to introduce this complexity. Have you excited something like MinVer[MinVer](https://github.com/adamralph/minver)? This problem has been solved with tools such as MinVer.

**Comment by ErikMogensen** [comment]
I haven't used Minver. It seems to be a good tool, but it looks to me that it will add complexity in this case since we still have to check if it is a prelease to determine if it should be published to Chocolatey and it requires changing all of the project files and adding one more component. 

To make changes to the build is time-consuming since a build takes 5 minutes and requires some gitting. That's a great advantage to PowerShell scripts which is just F5 and then it is running and allows debugging.

**Comment by SeanFeldman** [comment]
Manually tweaking the version and figuring out what it should be is complex. This is what MinVer and such already solved a long-time ago if you prefer to do it manually, by all means.

**Comment by SeanFeldman** [comment]
> I am not going to spend more time on this PR, unless someone start paying me for it.

☝️🙂

**Comment by ErikMogensen** [comment]
It is worth a try :wink:

**Review by SeanFeldman — APPROVED** [review]

**Pull request merged**